service/service.hpp

```diff
@@ -113,7 +113,7 @@ class Service : protected Program, public utility::Runnable {
 
     virtual int run() = 0;
 
-    virtual bool help(std::ostream &out, const std::string &what);
+    virtual bool help(std::ostream &out, const std::string &what) const;
```

Comment:
`override` would help here. Maybe we can mark overriden function with `override` here and in `cmdline` as well.

Reply:
done